x-pack/plugins/apm/server/routes/span_links/get_span_links_details.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { kqlQuery } from '@kbn/observability-plugin/server';
+import { chunk, isEmpty } from 'lodash';
+import {
+  SERVICE_NAME,
+  SPAN_ID,
+  SPAN_NAME,
+  TRACE_ID,
+  TRANSACTION_ID,
+  TRANSACTION_NAME,
+  TRANSACTION_DURATION,
+  SPAN_DURATION,
+  PROCESSOR_EVENT,
+  SPAN_SUBTYPE,
+  SPAN_TYPE,
+  AGENT_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { Environment } from '../../../common/environment_rt';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { SpanLinkDetails } from '../../../common/span_links';
+import { SpanLinks } from '../../../typings/es_schemas/raw/fields/span_links';
+import { SpanRaw } from '../../../typings/es_schemas/raw/span_raw';
+import { TransactionRaw } from '../../../typings/es_schemas/raw/transaction_raw';
+import { Setup } from '../../lib/helpers/setup_request';
+
+async function fetchSpanLinksDetails({
+  setup,
+  kuery,
+  spanLinks,
+}: {
+  setup: Setup;
+  kuery: string;
+  spanLinks: SpanLinks;
+}) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search('get_span_links_details', {
+    apm: {
+      events: [ProcessorEvent.span, ProcessorEvent.transaction],
+    },
+    _source: [
+      TRACE_ID,
+      SPAN_ID,
+      TRANSACTION_ID,
+      SERVICE_NAME,
+      SPAN_NAME,
+      TRANSACTION_NAME,
+      TRANSACTION_DURATION,
+      SPAN_DURATION,
+      PROCESSOR_EVENT,
+      SPAN_SUBTYPE,
+      SPAN_TYPE,
+      AGENT_NAME,
+    ],
+    body: {
+      size: 1000,
+      query: {
+        bool: {
+          filter: [
+            ...kqlQuery(kuery),
+            {
+              bool: {
+                should: spanLinks.map((item) => ({
+                  bool: {
+                    filter: [
+                      { term: { [TRACE_ID]: item.trace.id } },
+                      {
+                        bool: {
+                          should: [
```

Comment:
`minimum_should_match: 1` is implicit here, because there is no `filter` clause, but I think readability and durability would benefit from adding it explicitly (e.g., when someone adds a `filter` clause this will start returning different data.